File: src/dsa/recursion/printnumbers/print_numbers.c
#include <stdio.h>
void reversePrint(int num);
void print(int num)
{
    if (num == 0)
    {
        return;
    }
    print(num - 1);
    printf("%d ", num);
}

int main()
{
    printf("Enter number of elements: ");
    int num;
    scanf("%d", &num);
    reversePrint(num);
    printf("\n");
    print(num);
    return 0;
}

void reversePrint(int num)
{
    if (num == 0)
        return;
    printf("%d ", num);
    reversePrint(num - 1);
}

File: src/dsa/recursion/printnumbers/PrintNumbers.java
import java.util.Scanner;

public class PrintNumbers {
    public static void main(String args[]) {
        System.out.println("Enter number of elements: ");
        Scanner in = new Scanner(System.in);
        int num = in.nextInt();
        PrintNumbers pn = new PrintNumbers();
        pn.reversePrint(num);
        System.out.println();
        pn.print(num);
    }

    void reversePrint(int num) {
        if (num == 0) {
            return;
        }
        reversePrint(num - 1);
        System.out.print(num + " ");
    }

    void print(int num) {
        if (num == 0) {
            return;
        }
        System.out.print(num + " ");
        reversePrint(num - 1);
    }
}